### PR TITLE
dist/tools: extend the print_toolchain_version script

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

This PR adds some more tool version to be printed by the helper script. 

I think we might also want to make this output mandatory if someone reports an issue.

### Testing procedure

run `./dist/tools/ci/print_toolchain_versions.sh`

### CODE REVIEW

File: dist/tools/ci/print_toolchain_versions.sh [hunk 1]
```diff
@@ -8,6 +8,10 @@ get_cmd_version() {
     local cmd="$1"
     if command -v "$cmd" 2>&1 >/dev/null; then
         ver=$("$cmd" --version 2> /dev/null | head -n 1)
+        # some tools (eg. openocd) print version info to stderr
```
Comment (cladmi) [inline dist/tools/ci/print_toolchain_versions.sh]: Why not just get both output directly ?

Silencing stderr was done in your previous PR https://github.com/RIOT-OS/RIOT/commit/9968889c1302be783fd7cfffa9106279c7fee291 but the first version of the PR did not do it according to the code shown in https://github.com/RIOT-OS/RIOT/pull/8727#discussion_r171792583 and nothing explained why.

Also none of the previous code was redirecting `stderr` and it was introduced without justification in any commit or comment. Do you know why you added it ?

Comment (cladmi) [reply dist/tools/ci/print_toolchain_versions.sh]: I meant to put it on the line with the second `ver` calculation but somehow miss-clicked.

**Comment by smlng** [comment]
@cladmi I forgot a pending commit in the initial PR which I now included. It adds output to print OS and Kernel Version, which I think is useful for testing and reporting issues.

File: dist/tools/ci/print_toolchain_versions.sh [hunk 2]
```diff
@@ -30,6 +34,24 @@ get_define() {
...
+    printf "version=%s machine=%s processor=%s" $(uname -r) $(uname -m) $(uname -p)
+}
+
+get_os_info() {
+    local os="$(uname -s)"
+    local osname="unknown"
+    local osvers="unknown"
+    if [ "$os" == "Linux" ]; then
+        osname = "$(cat /etc/os-release | grep ^NAME= | awk -F'=' '{print $2}')"
+        osvers = "$(cat /etc/os-release | grep ^VERSION= | awk -F'=' '{print $2}')"
```
Comment (gebart) [inline dist/tools/ci/print_toolchain_versions.sh]: remove the spaces around the assignment operator. I get `./dist/tools/ci/print_toolchain_versions.sh: line 46: osname: command not found`

File: dist/tools/ci/print_toolchain_versions.sh [hunk 3]
```diff
@@ -30,6 +34,24 @@ get_define() {
     printf "%s" "$line"
 }
 
+get_kernel_info() {
+    printf "version=%s machine=%s processor=%s" $(uname -r) $(uname -m) $(uname -p)
```
Comment (gebart) [inline dist/tools/ci/print_toolchain_versions.sh]: Why not just `uname -a`?

Comment (gebart) [reply dist/tools/ci/print_toolchain_versions.sh]: Also, the uname calls must be quoted, I get this weird output on my machine:

```
Kernel: version=4.18.7-gentoo machine=x86_64 processor=Intel(R)version=Core(TM) machine=i7-6820HK processor=CPUversion=@ machine=2.70GHz processor=
```

File: dist/tools/ci/print_toolchain_versions.sh (hunk 2 above)
Comment (gebart) [reply dist/tools/ci/print_toolchain_versions.sh]: PRETTY_NAME might be enough. Below are some example os-release from various docker images:

```
# docker run -ti --rm fedora:latest cat /etc/os-release
NAME=Fedora
VERSION="28 (Twenty Eight)"
ID=fedora
VERSION_ID=28
PLATFORM_ID="platform:f28"
PRETTY_NAME="Fedora 28 (Twenty Eight)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:28"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=28
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=28
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"


# docker run -ti --rm debian:jessie cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"


# docker run -ti --rm debian:latest cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"


# docker run -ti --rm ubuntu:xenial cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial


# docker run -ti --rm ubuntu:latest cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic


# docker run -ti --rm alpine:latest cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.8.1
PRETTY_NAME="Alpine Linux v3.8"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL="http://bugs.alpinelinux.org"


# docker run -ti --rm gentoo/stage3-amd64 cat /etc/os-release
NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo/Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"


# docker run -ti --rm riot/riotbuild:latest cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

Comment (smlng) [reply dist/tools/ci/print_toolchain_versions.sh]: yes would work for (most) Linux Distros, but for MacOS I need to name and version separate - so it seems easiest to do so for Linux too.

File: dist/tools/ci/print_toolchain_versions.sh (hunk 3 above)
Comment (smlng) [reply dist/tools/ci/print_toolchain_versions.sh]: it now prints output of `uname -mprs` directly, `uname -a` includes the hostname which is not needed or might also compromise security/privacy if someone posts this output from their system in public.

**Comment by smlng** [comment]
comments addressed, its not really intrusive but actually is needed for proper test documentation. And as stated above, this script could also be handy for people reporting issues to post the output.

**Comment by smlng** [comment]
ah ... and may I squash?

**Comment by smlng** [comment]
@gebart all good now, so I can squash?

**Comment by smlng** [comment]
also looks nice in Murdock and Travis CI, and on macOS too:

```
Operating System Environment
-----------------------------
       Operating System: Mac OS X 10.13.6
                 Kernel: Darwin 17.7.0 x86_64 i386

Installed compiler toolchains
-----------------------------
             native gcc: Apple LLVM version 10.0.0 (clang-1000.10.44.2)
      arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
                avr-gcc: avr-gcc (GCC) 8.1.0
       mips-mti-elf-gcc: missing
             msp430-gcc: missing
   riscv-none-embed-gcc: missing
                  clang: Apple LLVM version 10.0.0 (clang-1000.10.44.2)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "2.5.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: "2.0.0" ("20150208")

Installed development tools
---------------------------
                  cmake: cmake version 3.12.1
               cppcheck: Cppcheck 1.84
                doxygen: 1.8.14
                 flake8: 3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 2.7.15 on Darwin
                    git: git version 2.18.0
                openocd: Open On-Chip Debugger 0.10.0
                 python: Python 2.7.15
                python2: Python 2.7.15
                python3: Python 3.7.0
             coccinelle: spatch version 1.0.6 compiled with OCaml version 4.05.0
```

**Pull request merged**